aes/src/lib.rs

```diff
@@ -136,24 +161,32 @@ cfg_if! {
         any(target_arch = "x86", target_arch = "x86_64"),
         not(aes_force_soft)
     ))] {
+        mod x86_64;
```

Comment:
The module is feature gated on `target_arch = "x86"` in addition to `x86_64` so perhaps the module name should just be `x86`?

That said I'm not sure we've actually successfully tested using AES-NI via a 32-bit build

Reply:
Yeah, good question, I wasn't sure about that either.

The reason I opted for `x86_64` though is because I was under the impression that the AES-NI extension was only actually available for `x86_64` architecture CPUs.

But from a brief google search it seems that it should be possible to get AES-NI to work for 32-bit targets though (based on some old Intel sample code).

Reply:
I believe it should still work when targeting those CPUs with a 32-bit binary, even if they're natively 64-bit

Reply:
Right.

I think I’d opt to keep it as `x86_64` since that’s the ISA where it’s available (even if running 32-bit target binaries), and also because 32-bit is slowly disappearing pretty much everywhere anyway.

But I don’t feel particularly strongly about it.

Would you like me to change it?

Reply:
I think it's a bit weird to say something is `x86_64` if it still works on a 32-bit target

Reply:
Could you check whether a 32 bit binary which uses VAES instructions works or not?

Reply:
> Could you check whether a 32 bit binary which uses VAES instructions works or not?

I just tested and it does work.

I used the following `.cargo/config.toml`:

```toml
[build]
target = "i686-unknown-linux-gnu"

[target.i686-unknown-linux-gnu]
linker = "clang-17"
rustflags = [
  "-C", "link-arg=-fuse-ld=lld-17",
  "-C", "link-arg=--target=i686-unknown-linux-gnu",
  "-C", "target-feature=+aes,+sse3,+vaes",
]
```

This is with `nightly-2024-07-02` on Ubuntu Mantic `23.10` using Debian multiarch configuration (with `i386` architecture added).

I also renamed the module to `x86`.

Reply:
I should note that I can't seem to actually test the 32-bit VAES target on CI though since I can't quite figure out how to get intel SDE to run those 32-bit binaries on the 64-bit host. It looks like it should be possible for it to run those but everything I've tried has resulted in it refusing to run them or just crashing (before getting to the tests).